lib/AST/ASTImporter.cpp

```diff
@@ -2775,7 +2775,9 @@ Decl *ASTNodeImporter::VisitFriendDecl(FriendDecl *D) {
   // Not found. Create it.
   FriendDecl::FriendUnion ToFU;
   if (NamedDecl *FriendD = D->getFriendDecl()) {
-    auto ToFriendD = cast_or_null<NamedDecl>(Importer.Import(FriendD));
+    auto *ToFriendD = cast_or_null<NamedDecl>(Importer.Import(FriendD));
+    if (ToFriendD && FriendD->getFriendObjectKind() != Decl::FOK_None)
+      ToFriendD->setObjectOfFriendDecl(true);
```

Comment:
Maybe i am missing something but I am not sure why it is needed. `setObjectOfFriendDecl` just sets up `Decl::IdentifierNamespace`, but that is copied anyway at the end of ASTImporter::Import(Decl*).

Reply:
Hmm, maybe you are right. The other change is what fixes the problem. This change is mainly to reduce the difference between our code and the version proposed for upstream: https://reviews.llvm.org/D32947

Reply:
Probably it is better to have this code here. It can be possible that `setObjectOfFriendDecl` sets new flags. It uses `getPreviousDecl` that can be different than in the to-be-imported version.